framework/test/mocha/common/utils/blocks_and_transactions_helper.js

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright © 2019 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const util = require('util');
+const Promise = require('bluebird');
```

Comment:
Why? Bluebird is still a better implementation than the native one - it's more flexible, has better performance and uses less memory.

References:
https://softwareengineering.stackexchange.com/a/279003
https://pub.clevertech.biz/native-javascript-promises-vs-bluebird-9e58611be22
https://github.com/petkaantonov/bluebird/tree/master/benchmark

Reply:
https://gist.github.com/joseluisq/a87adfd46d86d82ba787919a062f6e36
Those references are outdated with the recent node versions =)

Reply:
@shuse2 Not true, run the tests I referenced to. The ones you pointed to are testing only one case which is far from the normal application workload. Even if we treat them as meaningful - the difference is negligible, and we shouldn't drop Bluebird with all its features and flexibility because of micro-optimizations. We have much bigger problems to solve, bottlenecks are not here.

Reply:
I have no strict opinion on this topic. But what I can suggest is `bluebird` is a widely used and a good library with large user base. Using it in project should not be a problem, unless we have a clear benchmark available which justifies that using it cause problems to our product.

Reply:
I agree to refactor later in another issue, but one of our goals is to reduce 3rd party usage.
Therefore, we will remove the `bluebird` later along with `async` library.